### PR TITLE
Talha/pipfile precommit

### PR DESCRIPTION
**Pull request by mtalhatahir2001**
Created pipfile and added fastapi and precommit hook as dependency. Also configured the pre-commit-config.yaml file to run black, isort and flake8 before commit. 

Links to the tickets:
https://app.asana.com/0/1204411152293714/1204422722661444
https://app.asana.com/0/1204411152293714/1204422166915915

### CODE REVIEW

**Review by mustansirMuz — COMMENTED** [review]
Shouldn't there be individual config files for black and isort? Or does it work without them as well?

**Review by SohaibkhanFolio3 — APPROVED** [review]
Hmmm, Good work !!

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+repos:
+-   repo: https://github.com/pre-commit/mirrors-isort
+    rev: v4.3.21
+    hooks:
+    - id: isort
+-   repo: https://github.com/ambv/black
+    rev: stable
+    hooks:
+    - id: black
+      language_version: python3.6
```
Comment (RizwanNiaz) [inline .pre-commit-config.yaml:10]: Shouldn't this use python version 3.11 since that's the version specified in the pipfile?

Comment (mtalhatahir2001) [reply .pre-commit-config.yaml]: Yes, It should be 3.11, my bad totally forgot to change that.

**Review by mustansirMuz — APPROVED** [review]
Looks good.

**Pull request merged**